### ISSUE TITLE
Option to enforce newline before argument vector

### ISSUE THREAD

**Issue by vemv** [weavejester/cljfmt#240]
Hi,

cljstyle features an enforcement of a newline before a defn's argument vector:

```diff
- defn foo []
+ defn foo
+  []
```

While I don't necessarily agree with it, it seems a good idea to offer it for teams that value the practice.

PR welcome? Not sure if there's been a previous consideration.

**Comment by weavejester** [comment]
As long as it's optional and not the default, I'd welcome a PR.

**Comment by or** [comment]
Duplicate of #105.

**Comment by vemv** [comment]
I'll close this one.

Additional note, zprint tends to be particularly good at this sort of thing. I keep using cljfmt as a first pass in most projects.

**Comment by or** [comment]
Cool, I think I'll have a look at adding this sometime soon, it definitely makes sense as an option.

I'm just grooming the issues a bit. :) Maybe react on the other issue, if you want to be kept in the loop.

**Comment by jakepearson** [comment]
@or Are you still considering implementing this feature? My team would love to use it. 😄

**Comment by vemv** [comment]
@jakepearson FYI zprint supports this, and closely related, opt-in styling. Its author has been generous and communicative with me implementing a bespoke set of 'rules'.

Feel free to check out open/closed issues in zprint created by me. We called 'rules of defn' or `:rod`, with different, atomic rules like `rod1`, `rod2`, etc. 

I had it pending to give one last round of feedback, but well, I never find the time...

Personally keep using cljfmt daily, I've used zprint for specific professional jobs, or for tooling research.
